x/burn/module.go

```diff
@@ -0,0 +1,118 @@
+package burn
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/medibloc/panacea-core/x/burn/internal/keeper"
+
+	"github.com/cosmos/cosmos-sdk/x/supply"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/gorilla/mux"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/medibloc/panacea-core/x/burn/internal/types"
+)
+
+type AppModuleBasic struct{}
+
+type AppModule struct {
+	AppModuleBasic
+	keeper       keeper.Keeper
+	supplyKeeper supply.Keeper
+}
+
+var (
+	_ module.AppModuleBasic = AppModuleBasic{}
+	_ module.AppModule      = AppModule{}
+)
+
+func NewAppModule(keeper keeper.Keeper, supplyKeeper supply.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{},
+		keeper:         keeper,
+		supplyKeeper:   supplyKeeper,
+	}
+}
+
+func (AppModuleBasic) Name() string {
+	return types.ModuleName
+}
+
+func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
+	types.RegisterCodec(cdc)
+}
+
+func (AppModuleBasic) DefaultGenesis() json.RawMessage {
+	return types.ModuleCdc.MustMarshalJSON(types.DefaultGenesisState())
+}
+
+func (AppModuleBasic) ValidateGenesis(bz json.RawMessage) error {
+	var data types.GenesisState
+	if err := types.ModuleCdc.UnmarshalJSON(bz, &data); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
+	}
+
+	return types.ValidateGenesis(data)
+}
+
+func (AppModuleBasic) RegisterRESTRoutes(_ context.CLIContext, _ *mux.Router) {}
+
+func (AppModuleBasic) GetTxCmd(_ *codec.Codec) *cobra.Command {
+	return nil
+}
+
+func (AppModuleBasic) GetQueryCmd(_ *codec.Codec) *cobra.Command {
+	return nil
+}
+
+func (am AppModule) InitGenesis(ctx sdk.Context, data json.RawMessage) []abci.ValidatorUpdate {
+	ctx.Logger().Info("call init genesis")
+	var genesisState types.GenesisState
+	types.ModuleCdc.MustUnmarshalJSON(data, &genesisState)
+	keeper.InitGenesis(ctx, am.keeper, genesisState)
+	return []abci.ValidatorUpdate{}
+}
+
+func (am AppModule) ExportGenesis(ctx sdk.Context) json.RawMessage {
+	ctx.Logger().Info("call export genesis")
+	gs := keeper.ExportGenesis(ctx, am.keeper)
+	return types.ModuleCdc.MustMarshalJSON(gs)
+}
+
+func (AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {
+
+}
+
+func (AppModule) Route() string {
+	return ""
+}
+
+func (AppModule) NewHandler() sdk.Handler {
+	return nil
+}
+
+func (AppModule) QuerierRoute() string {
+	return ""
+}
+
+func (_ AppModule) NewQuerierHandler() sdk.Querier {
+	return nil
+}
+
+func (AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
+
+func (am AppModule) EndBlock(ctx sdk.Context, block abci.RequestEndBlock) []abci.ValidatorUpdate {
+	err := am.keeper.BurnCoins(ctx, types.BurnAddress)
+
+	if err != nil {
+		panic("Failed burn. msg : " + err.Error())
```

Comment:
I'm not sure if it's okay to panic in the `EndBlock`. Let me google it.

Reply:
@youngjoon-lee 
I think... `panic` doesn't seem to be a good way.

Reply:
Delete "panic" and modify to write error log.